### PR TITLE
implementation and demo for json_table.json_table() functionality modelled after Oracle 12c json_table()

### PR DESCRIPTION
**Pull request by dsnz**
implementation and demo for json_table.json_table() functionality modelled after Oracle 12c json_table()

### CODE REVIEW

File: src/pljson_table_impl_body.typ [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    --row_inds index_array := index_array();
+    j number;
+    num_rows number := 0;
+  begin
+    --dbms_output.put_line('>>Fetch');
+    
+    anydataset.begincreate(dbms_types.typecode_object, self.ret_type, outset);
+    
+    /* iterative cartesian product algorithm */
+    <<main_loop>>
```
Comment (jsumners) [inline src/pljson_table_impl_body.typ]: I haven't seen this sort of operation before. Is it meant to be a comment?

**Comment by dsnz** [comment]
Hi,

I'm not sure about which operation you mean. I guess it's "<<main_loop>>" which is not a comment but a label for the loop and is used later to specify exactly the loop to exit from.

**Comment by jsumners** [comment]
I see. Thanks.

**Comment by jsumners** [comment]
I think this should be moved to the "optional" section of the install script.

**Comment by dsnz** [comment]
Hi,

I never thought any part of pljson as optional as all are installed by a user usually without any choice (and after all they do no harm or consume space). I felt this is a unnecessary leftover from some very old installation but anyway I will move json_table to this optional section.

**Comment by jsumners** [comment]
They're installed by default, but if the installer so chooses, they can comment out each of them without affecting the core functionality of PL/JSON.

**Comment by dsnz** [comment]
Ok, moved to optional section.

**Pull request merged**